### ISSUE TITLE
Enamorus Doesn't Pass Legality Check (Pokemon Legends Arceus)

### ISSUE THREAD

**Issue by TankedThomas** [kwsch/PKHeX#4343]
**Describe the problem**
Freshly-caught Enamorus from PLA does not pass legality checks.
The error is: `Invalid: Unable to match an encounter from origin game.`

**To Reproduce**
This is the unmodified pa8 file extracted immediately after catching Enamorus: [0905 - Enamorus - E238A763231E.pa8.txt](https://github.com/user-attachments/files/16646944/0905.-.Enamorus.-.E238A763231E.pa8.txt)

**Expected behavior**
This encounter should be valid. Without further information from PKHeX about the error, I can't be sure why it says it isn't.

**Screenshots**
![image](https://github.com/user-attachments/assets/f4d3bd93-957a-4ec4-8301-d78cec045702)
![image](https://github.com/user-attachments/assets/bf58ee89-3aa4-4291-b768-5de73fad224d)

**Additional context**
It was obtained legitimately without any cheating. My Switch is obviously modified to be able to dump the save, but other than that, everything is stock.

**Comment by Lusamine** [comment]
The issue is met location of "Crimson Mirelands" which is the general area when you're not in a specific area. PKHeX currently expects "Scarlet Bog". Do you remember where exactly on the map you caught it from?

**Comment by TankedThomas** [comment]
I think it was in Scarlet Bog, where it's initially found. Either way, I didn't modify it at all, so the game is responsible for that met location.

**Comment by Lusamine** [comment]
"Scarlet Bog" is a really big area and not very specific. I'm wondering if you can point out where on the map you caught it at so I can try it out as well to fix this.

**Comment by kwsch** [comment]
![map_lmap_pic_l_01](https://github.com/user-attachments/assets/bcbeee23-9fa3-4d42-97ca-2d06725199a3)

**Comment by Lusamine** [comment]
^If you can mark where on the map you were with an X, or provide an image from in-game, that would help a lot more than "Scarlet Bog ... or slightly outside there" which is a very big area to track down.

**Comment by TankedThomas** [comment]
I don't have any screenshots or anything so I can't really remember for sure since it was a day or two ago. The best I could do is try to replicate it with a previous save.

I booted up the game to try to refresh my memory but I can't be sure. Off the top of my head, I think it was somewhere around here:
![358880002-bcbeee23-9fa3-4d42-97ca-2d06725199a3](https://github.com/user-attachments/assets/007df4e1-cbc7-4d08-ab66-18c1012740ee)

I did just try catching some Pokemon around this area, and a Geodude caught right around here has "Crimson Mirelands" as its met location, so this seems about right.

**Comment by Lusamine** [comment]
Standing anywhere outside of those yellowish boxes would give "Crimson Mirelands". I'll try tomorrow to see if it's possible to chase Enamorus over there. Do you remember if you threw the ball from far away, if you did it in battle, or if you threw the ball and then ran before it finished catching?

**Comment by TankedThomas** [comment]
I definitely battled it. I was using Feather Balls to take down its shield, then started a battle with it to catch it (and caught it during the battle). I was very close to it when I started the battle.

**Comment by Lusamine** [comment]
That helps to know you did it in battle and didn't intentionally do anything too extraordinary. Some of the static legend crossovers have freaky conditions like Cresselia (get in battle, run towards the exit while in battle, throw the ball and keep running while it catches) or Darkrai (throw a ball at it and them rapidly jump/fly over the cliff into the adjacent area).

**Comment by Lusamine** [comment]
I was able to catch Enamorus all the way up the hill northeast of Scarlet Bog to get the "Bolderoll Slope" met location, and the generic "Crimson Mirelands" is much easier than that.

![image](https://github.com/user-attachments/assets/5b701ed0-0487-4a9d-8f8f-b183a94a3c43)

Since it was possible with Gigaton Ball, we expect the other balls to be possible too.

Here's a video if you're curious how this was done: https://streamable.com/ou7lmv
It was fog weather so it's hard to see. I stunned it twice, chased it around to the base of the hill, then approached it from the hill side to stun the 3rd time, throw a ball, and run back up the hill.
